src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesGraphProvider.cs

```diff
@@ -215,41 +217,34 @@ Graph IGraphProvider.Schema
                     return;
                 }
 
-                // All graph nodes generated here will have this unique node id, root node will have it equal to null
-                var nodeIdString = GetPartialValueFromGraphNodeId(inputGraphNode.Id, CodeGraphNodeIdName.File);
-                var nodeId = DependencyNodeId.FromString(nodeIdString);
-                if (nodeId == null)
-                {
-                    continue;
-                }
-
                 var projectPath = GetPartialValueFromGraphNodeId(inputGraphNode.Id, CodeGraphNodeIdName.Assembly);
                 if (string.IsNullOrEmpty(projectPath))
                 {
                     continue;
                 }
 
-                var subTreeProvider = await GetSubTreeProviderAsync(graphContext, 
-                                                                    inputGraphNode, 
-                                                                    projectPath,
-                                                                    nodeId).ConfigureAwait(false);
-                if (subTreeProvider == null)
+                var nodeInfo = await GetDependencyNodeInfo(graphContext, inputGraphNode, projectPath).ConfigureAwait(false);
```

Comment:
We can use C# 7 tuples here. So this can simply become 

```csharp
var (node, subTreeProvider) = await GetDependencyNodeInfo(graphContext, inputGraphNode, projectPath).ConfigureAwait(false); 
```

You'll need https://github.com/dotnet/roslyn-project-system/pull/756 to be merged in first though. @333fred